### PR TITLE
Remove graphql-yoga

### PR DESCRIPTION
**Pull request by Tirokk**
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-13T16:46:32Z" title="Saturday, July 13th 2019, 6:46:32 pm +02:00">Jul 13, 2019</time>_
_Merged <time datetime="2019-07-16T07:13:45Z" title="Tuesday, July 16th 2019, 9:13:45 am +02:00">Jul 16, 2019</time>_
---

## 🍰 Pullrequest
After we decided to seed our database directly in our tests and not going over HTTP, this PR will remove `graphql-yoga` (insufficiently maintained, 3rd party) for `apollo-server`. This will also free us from the obsolete extra server during backend testing, `apollo-server-testing` works without starting an entire HTTP server. We even might get reasonable code coverage now @ulfgebhardt :wink: 

doge

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-07-13T17:21:55Z" title="Saturday, July 13th 2019, 7:21:55 pm +02:00">Jul 13, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/1040?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@4c91d8f`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

File: backend/src/server.js [hunk 1]
```diff
@@ -20,28 +20,30 @@ const driver = getDriver()
 
 const createServer = options => {
   const defaults = {
-    context: async ({ request }) => {
-      const user = await decode(driver, request.headers.authorization)
+    context: async ({ req }) => {
```
Comment (Tirokk) [inline backend/src/server.js:23]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-15T10:41:00Z" title="Monday, July 15th 2019, 12:41:00 pm +02:00">Jul 15, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -20,28 +20,30 @@ const driver = getDriver()
 
 const createServer = options => {
   const defaults = {
-    context: async ({ request }) => {
-      const user = await decode(driver, request.headers.authorization)
+    context: async ({ req }) => {
```


Can you explain this change?
I see cases where I would like to go through the code base and do the opposite where I spell it  out further, but I respect that others like to shorten variables, but if I lengthen and you shorten, we are kinda working against each other, or should we try to decide a standard and all stick by it?

Comment (Tirokk) [reply backend/src/server.js:23]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-15T11:20:56Z" title="Monday, July 15th 2019, 1:20:56 pm +02:00">Jul 15, 2019</time>_

---


Same, same … 😂

Comment (Tirokk) [reply backend/src/server.js:23]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-15T15:45:37Z" title="Monday, July 15th 2019, 5:45:37 pm +02:00">Jul 15, 2019</time>_

---


Again, not my decision, but `apollo`'s decision. I would spell it `request` and so does `graphql-yoga`, but `apollo` uses `req`.

Comment (Tirokk) [reply backend/src/server.js:23]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-15T15:48:40Z" title="Monday, July 15th 2019, 5:48:40 pm +02:00">Jul 15, 2019</time>_

---


Read [this](https://github.com/prisma/graphql-yoga/issues/449#issuecomment-440595783) for the **why**.

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-15T16:13:10Z" title="Monday, July 15th 2019, 6:13:10 pm +02:00">Jul 15, 2019</time>_

---


@mattwr18 @Tirokk ES2015 template literals: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: backend/src/jest/helpers.js [hunk 2]
```diff
@@ -15,3 +15,9 @@ export async function login(variables) {
     authorization: `Bearer ${response.login}`,
   }
 }
+
+//* This is a fake ES2015 template string, just to benefit of syntax
+// highlighting of `gql` template strings in certain editors.
+export function gql(strings) {
+  return strings.join('')
+}
```
Comment (Tirokk) [inline backend/src/jest/helpers.js:23]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-15T19:48:51Z" title="Monday, July 15th 2019, 9:48:51 pm +02:00">Jul 15, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -15,3 +15,9 @@ export async function login(variables) {
     authorization: `Bearer ${response.login}`,
   }
 }
+
+//* This is a fake ES2015 template string, just to benefit of syntax
+// highlighting of `gql` template strings in certain editors.
+export function gql(strings) {
+  return strings.join('')
+}
```


I like this solution, but how do you feel about it @roschaefer 
don't want you to feel like you are making too many concessions
I am also ok dealing with no syntax highlighting, not sure how @Tirokk feels.

File: backend/src/server.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/server.js:23]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-16T07:06:50Z" title="Tuesday, July 16th 2019, 9:06:50 am +02:00">Jul 16, 2019</time>_

---


Thanks for letting us understand. 👍🏼

**Review by Mogge — APPROVED** [review]

**Review by Mogge — CHANGES_REQUESTED** [review]
I always implement the `gql` tag and we have just spoken about that I would love to have that for Cypher as well.
So I really don’t understand, why you remove it … @roschaefer 
Please let it in place!

Your work at all seems to be great here, @roschaefer ! 🚀💫

**Review by Mogge — APPROVED** [review]
Cool you found that out @roschaefer ! 😎
Works absolut fine with me. 😁

Nothing is written in stone

**Pull request closed without merging**